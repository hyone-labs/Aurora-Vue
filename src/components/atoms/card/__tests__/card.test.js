import { render } from '@testing-library/vue';
import HCard from '../card.vue';

describe('HCard', () => {
  it('render correctly with default props', () => {
    const { html } = render(HCard, {
      slots: {
        default: 'My first test',
      },
    });
    expect(html()).toMatchSnapshot();
  });

  it('render correctly with variant', () => {
    const { container } = render(HCard, {
      props: {
        variant: 'primary',
      },
      slots: {
        default: 'My second test',
      },
    });
    expect(container.firstChild.classList.contains('h-card--variant-primary')).toBe(true);
  });

  it('render correctly with elevation', () => {
    const { container } = render(HCard, {
      props: {
        elevation: 'high',
      },
      slots: {
        default: 'My second test',
      },
    });
    expect(container.firstChild.classList.contains('h-card--elevation-high')).toBe(true);
  });

  it('render correctly with elevation', () => {
    const { container } = render(HCard, {
      props: {
        tag: 'section',
      },
      slots: {
        default: 'My second test',
      },
    });
    expect(container.firstChild.tagName).toBe('SECTION');
  });
});
