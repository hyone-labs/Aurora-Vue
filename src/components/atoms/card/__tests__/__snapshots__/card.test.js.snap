// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HCard > render correctly with default props 1`] = `"<div class="h-card h-card--color-default h-card--variant-default h-card--elevation-medium h-card--behavior-rounded">My first test</div>"`;
